alacritty_terminal/src/grid/row.rs

```diff
@@ -28,13 +30,33 @@ impl<T: PartialEq> PartialEq for Row<T> {
     }
 }
 
-impl<T: Copy> Row<T> {
-    pub fn new(columns: Column, template: T) -> Row<T>
+impl<T: Clone> Row<T> {
+    /// Create a new terminal row.
+    ///
+    /// Ideally the `template` should be `Copy` in all performance sensitive scenarios.
+    pub fn new<I>(columns: Column, template: I) -> Row<T>
     where
+        I: Into<T> + Clone,
         T: GridCell,
     {
-        let occ = if template.is_empty() { 0 } else { columns.0 };
-        Row { inner: vec![template; columns.0], occ }
+        debug_assert!(columns.0 >= 1);
```

Comment:
This should always be guaranteed, due to our minimum grid size. I do not want to waste performance on an assertion that will never be hit anyways.

Though if it ever gets past this, it will crash terribly.